lib/plugin.js

```diff
@@ -1,13 +1,16 @@
 'use strict'
 
 const createError = require('http-errors')
+const parseSignatureHeader = require('./parseSignatureHeader')
+const extractSignature = require('./extractSignature')
 
 function plugin (pluginOptions) {
   const defaultOptions = {
     sharedSecret: null,
     verificationError: 'Unauthorized',
     verificationErrorMessage: 'Signature verification failed',
-    extractSignature: () => {},
+    parseSignatureHeader,
```

Comment:
I don't think we actually want to expose this option to the user directly. That way we keep the API generic and flexible. There are plenty of HMAC signed request implementations that don't use the `Signature` header (some use `Authorization` for example). We should allow configuration to support for those 👍

Reply:
Okay, so I think I get what we're after, here.
I had been going down a route sort of like this:
```js
function plugin() {
  const defaultOptions = {
    parseSignatureHeaders,
    pullSignatureBase64String,
    doSomethingWithString,
  };
  function verificationHook(req, reqply, next) {
    parseSignatureHeaders(req)
      .then(pullSignatureBase64String)
      .then(doSomethingWithString)
      .then(next);
  }
}
```
Where we expect people to overload whichever steps with their own config that we need.

When what I should be doing is closer to this:
```js
function plugin() {
  const defaultOptions = {
    extractSignature,
  };
  function verificationHook(req, reqply, next) {
    extractSignature(req)
    next()
  }
}

function extractSignature (req) {
  fp.pipeline(req [
    parseSignatureHeaders,
    pullSignatureBase64String,
    doSomethingWithString,
  ])
}
```
Where I write a generic extractSignature function that does all of the transforms with the assumption that what you're passing in is a `{Header: Signature: ...}` style request.